client_update.go

```diff
@@ -0,0 +1,183 @@
+// Copyright 2016 Mender Software AS
+//
+//    Licensed under the Apache License, Version 2.0 (the "License");
+//    you may not use this file except in compliance with the License.
+//    You may obtain a copy of the License at
+//
+//        http://www.apache.org/licenses/LICENSE-2.0
+//
+//    Unless required by applicable law or agreed to in writing, software
+//    distributed under the License is distributed on an "AS IS" BASIS,
+//    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//    See the License for the specific language governing permissions and
+//    limitations under the License.
+package main
+
+import (
+	"encoding/json"
+	"io"
+	"io/ioutil"
+	"net/http"
+
+	"github.com/mendersoftware/log"
+	"github.com/pkg/errors"
+)
+
+const (
+	minimumImageSize int64 = 4096 //kB
+)
+
+type Updater interface {
+	GetScheduledUpdate(server string, deviceID string) (interface{}, error)
+	FetchUpdate(url string) (io.ReadCloser, int64, error)
+}
+
+type UpdateClient struct {
+	client       *http.Client
+	minImageSize int64
+}
+
+func NewUpdateClient(conf httpsClientConfig) (*UpdateClient, error) {
+	client, err := NewHttpClient(conf)
+	if client == nil {
+		return nil, errors.Wrap(err, "failed to create updater HTTP client")
+	}
+
+	up := UpdateClient{
+		client:       client,
+		minImageSize: minimumImageSize,
+	}
+	return &up, nil
+}
+
+func (u *UpdateClient) GetScheduledUpdate(server string, deviceID string) (interface{}, error) {
```

Comment:
The idea of adding "processing function" as a parameter was to avoid tight coupling and rather use dependency injection. As you are simplifying this, which I think makes sense, maybe we can get rid of `getUpdateInfo()` parameter as well and rather have `GetScheduledUpdate()`  interface implementation for different clients instead.